libraries/SITL/SIM_Gazebo.cpp

```diff
@@ -130,6 +147,7 @@ void Gazebo::update(const struct sitl_input &input)
 
     // update magnetic field
     update_mag_field_bf();
+    drain_sockets();
```

Comment:
Background for this can be found here: https://github.com/osrf/uctf/issues/46

There can be issues under high load or lossy communications where multiple incoming messages can arrive. And if you get into that state without draining the incoming queue you end up reading the old packets, which effectively causes phase lag in your control loop.